frontend/src/components/user-administration/user-administration-modal/user-administration-modal.tsx

```diff
@@ -0,0 +1,278 @@
+    lastname: z.string().min(1),
+    email: z.string().email()
+  });
+
+  const getRoles = (): string[] | null => {
+    try {
+      fetch('/backend/users/roles', {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: `Bearer ${user.token}`
+        }
+      }).then(async (response) => {
+        const rolesData = await response.json();
+        const tempRoles: any[] = [];
+
+        rolesData.forEach((role) => {
+          tempRoles.push(
+            <MenuItem value={role} key={role}>
+              {role}
+            </MenuItem>
+          );
+        });
+        setRoles(tempRoles);
+      });
+    } catch (error: any) {
+      toast.error(error.message);
+      return null;
+    }
+    return null;
+  };
+
+  const customErrorMap = () => {
+    return (issue: ZodIssueOptionalMessage, ctx: ErrorMapCtx) => {
+      if (
+        issue.code === z.ZodIssueCode.invalid_string ||
+        issue.code === z.ZodIssueCode.too_small
+      ) {
+        if (issue.path.includes('firstname')) {
+          return {
+            message: `Bitte geben Sie einen gültigen Vornamen ein.`
+          };
+        }
+        if (issue.path.includes('lastname')) {
+          return {
+            message: `Bitte geben Sie einen gültigen Nachnamen ein.`
+          };
+        }
+        if (issue.path.includes('email')) {
+          return {
+            message: `Bitte geben Sie eine gültige E-Mail ein.`
+          };
+        }
+      }
+      return { message: ctx.defaultError };
+    };
+  };
+
+  const { handleSubmit, control } = useForm({
+    resolver: zodResolver(formSchema, {
+      errorMap: customErrorMap()
+    }),
+    mode: 'onSubmit',
+    defaultValues: {
+      firstname: '',
+      lastname: '',
+      email: '',
+      roles: ''
+    }
+  });
+
+  const handleFormSubmit = async (data: {
+    firstname: string;
+    lastname: string;
+    email: string;
+  }) => {
+    const body = {
+      name: data.firstname,
+      surname: data.lastname,
+      email: data.email,
+      role: role
+    };
+
+    try {
+      await fetch('/backend/users/signup', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: `Bearer ${user.token}`
+        },
+        body: JSON.stringify(body)
+      }).then(async (res: any) => {
+        if (res.ok) {
+          const data: UserDto = await res.json();
+          onAddedUser(data);
+          setPassword(data.password);
+          setOpenShowPasswordModal(true);
+        }
+      });
+    } catch (error: any) {
+      toast.error(error.message);
+    }
+  };
+
+  const handleRoleChange = (event: SelectChangeEvent<typeof role>) => {
+    setRole(event.target.value);
+  };
+
+  const closePasswordModal = () => {
+    control._reset();
+    setRole(null);
+    setShowModal(false);
+  };
+
+  const [roles, setRoles] = useState<any[] | null>(() => getRoles());
```

Comment:
I think we have a Enum for the roles, and avoid using "null" as state, you can check in the setter, 

if(value){
setRoles(value)
}

then you don't have "nulls"

UserRole

Reply:
Es gibt nur noch eine Rolle! Entweder Admin oder user

Reply:
Thank you for the enum hint.

The value of the role is null, when you want to add a new user because we don't have any default values, do we? The role needs to be emptied/nulled on closing, otherwise the field will be refilled with the previous selection upon creating an additional user. Do you know what i mean?